supabase/migrations/20260503000100_backfill_exercise_ids.sql

```
@@ -0,0 +1,78 @@
+-- =============================================================
+
+-- routine_exercises
+UPDATE routine_exercises re
+SET exercise_id = ec.id
+FROM exercise_catalog ec
+WHERE re.exercise_id IS NULL
+  AND LOWER(TRIM(re.name)) = LOWER(TRIM(ec.name))
+  AND LOWER(TRIM(ec.name)) IN (
+      SELECT LOWER(TRIM(name)) FROM exercise_catalog
+      WHERE is_custom = FALSE
+      GROUP BY LOWER(TRIM(name)) HAVING COUNT(*) = 1
```

SQLFluff (4.1.0)

[error] The 'HAVING' keyword should always start a new line.

(LT14)